collect_app/src/main/java/org/odk/collect/android/version/VersionInformation.java

```diff
@@ -0,0 +1,75 @@
+package org.odk.collect.android.version;
+
+import androidx.annotation.Nullable;
+
+public class VersionInformation {
```

Comment:
This is much better. It very nearly could close https://github.com/getodk/collect/issues/3693 if you could add the couple of test cases suggested. I still think the regex approach could cut some of this down but it's now much easier to reason about so I'm fine with it as-is.

Reply:
Ah didn't see that issue. Definitely the right time. Will get to that once other issues are fixed!

Reply:
Looks like the implementation was there for this but wanted to move the coverage from the ViewModel to the `VersionInformation` so did that.